lib/parseSty.js

```diff
@@ -0,0 +1,177 @@
+ * @returns {Array[]} An array of bracket sections, each containing a body and the type of bracket.
+ */
+var findBracketSections = module.exports.findBracketSections = function(string, sections) {
+    if (!sections) {
+        sections = [];
+    }
+    var bracket = balanced("[", "]", string);
+    var brace = balanced("{", "}", string);
+    if (brace && (!bracket || brace.start < bracket.start)) {
+        sections.push([brace.body, "{"]);
+        return findBracketSections(brace.post, sections);
+    }
+    else if (bracket) {
+        sections.push([bracket.body, "["]);
+        return findBracketSections(bracket.post, sections);
+    }
+    return sections;
+};
+
+/**
+ * Finds the first occurrence in the provided string of the earliest occurring item in the given array, starting from
+ * a specified index. Returns both the index of occurrence and the relevant item of subs (returns -1 and the first item
+ * if none of the provided substrings can be found).
+ *
+ * @param {string} search - The string to search.
+ * @param {string[]} substrings - All items to search for.
+ * @param {number} start - The index to start searching at.
+ * @returns {Array} The index of the first matching item and the item itself as an array.
+ */
+var findFirst = module.exports.findFirst = function(search, substrings, start) {
+    var min = -1;
+    var index;
+    var first = substrings[0];
+    for (var i in substrings) {
+        index = search.indexOf(substrings[i], start);
+        if ((index !== -1) && (index < min || min === -1)) {
+            min = index;
+            first = substrings[i];
+        }
+        if (min === 0) {
+            break;
+        }
+    }
+    return [min, first];
+};
+
+/**
+ * Parses a command declaration of form \newcommand (or \renewcommand) followed by a command and its replacement. If the
+ * declaration is valid, an association is added in cmds between the command string and its replacement. Otherwise, an
+ * error is thrown stating that the declaration is invalid and also providing the declaration in question.
+ *
+ * @param {string} declaration - The replacement declaration.
+ * @param {string} command - The command used to call the declaration (\newcommand or \renewcommand).
+ * @param {Object} cmds - An associative array between commands and their respective replacements.
+ */
+function parseNewCommand(declaration, command, cmds) {
+    var definition = declaration.substring(command.length);
+    var sections = findBracketSections(definition);
+    switch (sections.length) {
+        // \newcommand\command\replacement
+        case 0:
+            // /\\[a-zA-Z]+/ depends on the alphabetic restrictions on a LaTeX command name.
+            var commands = definition.match(/\\[a-zA-Z]+/g);
+            cmds[commands[0]] = commands[1];
+            break;
+        case 1:
+            var cmd = definition.replace(new RegExp(sections[0][0].replace(/\\/g, "\\\\"), 'g'), "").match(/\\[a-zA-Z]+/)[0];
```

Comment:
Maybe it would make sense to introduce a a variable for sections[0] ?
Would that be \command1,'{' int the example below.